Feature/login

TimvdLippe opened the pull request.

Commit:
Initial implementation of authentication

TimvdLippe committed:
Enable user authenticating via token

jwgmeligmeyling commented on src/main/java/nl/tudelft/planningstool/api/security/AuthorizationFilter.java:
```diff
@@ -0,0 +1,84 @@
+    // Extract the roles from the annotated element
+    private List<Role> extractRoles(AnnotatedElement annotatedElement) {
+        if (annotatedElement == null) {
+            return new ArrayList<Role>();
```
`Collections.emptyList()` is cheaper to instantiate (and in the JDK).

jwgmeligmeyling commented:
Implementation of the `AuthenticationFilter` looks good! Seems the added dependency to `jetty-security` is still unused, are you planning to use it in the future? As it seems you're planning to do token based authentication (by query parameter?) I don't think you actually need it. Keep in mind that when doing token based authentication users should not be allowed to exchange urls or they will give away their identity (if it's internal API or this constraint is very clear there is no problem). Also be sure that the token generation is relatively strong (properly seeded rng)

TimvdLippe commented:
The tokens are only user in client-server communication, so the users should not see anything directly. The dependency was added by Gijs, he knows if we need it later or not.

I think the commented code was the generator code, is that secure enough?

jwgmeligmeyling commented:
Ah I missed that code. SecureRandom as basis for a (relatively long) string should be enough!

GijsWeterings commented:
`jetty-security` is leftover from first implementation. I'll nuke that :shipit:

TimvdLippe committed:
Add uuid field to TokenResponse and enable token generator

Commit:
removed unused dependency

Commit:
Merge branch 'feature/login' of github.com:Mentoraat/plannings-tool-server into feature/login

jwgmeligmeyling commented:
:+1:

TimvdLippe committed:
Edit database name in persistence properties

TimvdLippe committed:
Add register end-point for authentication

TimvdLippe committed:
Add metamodel folder back

TimvdLippe committed:
Security updates

- Return Collections.emptyList();
- Remove accesstoken and uuid from UserResponse
- Move target/metamodel as source folder instead of resource folder

TimvdLippe committed:
Add indexes and tidy up some methods
